src/Common/Simulation.cpp

```diff
@@ -188,6 +188,7 @@ SimulationBase::SimulationBase
 #else
   Nthreads = 1;
 #endif
+  timing = new CodeTiming() ;
```

Comment:
We could do. My reason for doing it this way is that in ProcessParameters etc we tend to copy the timing pointer. So adding it to the destructor meant changing fewer lines of code.

Also, we always interact with the timing object through one of those such pointers, that are always called timing. By using the same name here for the master one then we can always call the timing object in exactly the same way, which sould be less error prone.